src/cbc_sdk/platform/devices.py

```diff
@@ -485,6 +486,24 @@ def set_deployment_type(self, deployment_type):
         self._update_criteria("deployment_type", deployment_type)
         return self
 
+    def set_max_rows(self, max_rows):
+        """
+        Sets the max number of devices to fetch in a singular query
+
+        Args:
+            max_rows (integer): Max number of devices
+
+        Returns:
+            DeviceSearchQuery: This instance.
+
+        Raises:
+            ApiError: If rows is negative or greater than 10000
+        """
+        if max_rows < 0 or max_rows > 10000:
+            raise ApiError("Max rows must be between 0 and 10000")
```

Comment:
What's the benefit of setting `max_rows = 0`? An API response with no returned rows - does that do anything practically useful, or just theoretically interesting?

Reply:
It's a supported value and could be used to validate a search or check num_found without fetching any results. May not be practical but it's not something I want to prevent